### ISSUE TITLE
Retry failed self-update

### ISSUE THREAD

**Issue by Gerrit-K** [rancher-sandbox/rancher-desktop#1929]
### Problem Description

The auto update failed on my machine (x86 mac) a lot of times without me noticing it. Upon inspecting the logs I noticed this error:

```
2022-03-30T10:21:39.999Z: Error: Error: net::ERR_NETWORK_CHANGED
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:101:7068)
    at SimpleURLLoaderWrapper.emit (node:events:394:28)
```

The reason for this network change is (allegedly) RD messing up my VPN on startup, causing a reconnect, but that's probably worth a separate issue. However, once the update failed, there's no way to restart the self-update, other than restarting Rancher Desktop entirely. In my special case, though, this again results in a VPN reconnect, leading to another failed update, so I'm stuck in this loop and have to manually update.

### Proposed Solution

It would be nice if there was a button to simply retry checking for updates and applying them. Alternatively it would also be nice to automatically retry in case of "retryable" errors, such as ERR_NETWORK_CHANGED, or 503 for example.

### Preflight Checklist

- [X] I have searched the [issue tracker](https://github.com/rancher-sandbox/rancher-desktop/issues) for a feature request that matches the one I want to file, without success.

**Comment by jandubois** [comment]
I'm curious how/why Rancher Desktop would cause a VPN reconnect; can you provide more information about this?

**Comment by Gerrit-K** [comment]
@jandubois I can try. I'm on macOS 12.3 (x86) and the VPN software I'm using is Pulse Secure. It's configured to automatically start on login and so is Rancher Desktop. By the time RD hits the `sudo` prompt, the VPN is usually already connected. I can't pinpoint the exact action, but right before or at the beginning of "Starting virtual machine", the VPN loses connection for a second and tries to reconnect. 

My assumption is that RD is changing some network config (e.g. adding a bridge interface or sth) and the VPN software is listening on those changes and then reconnects. But I have no evidence for that. If you can give me a hint on where to look and how to get more information, I can try to provide more details.

**Comment by jandubois** [comment]
> Regardless of the root cause of this issue, is there a way to 'Check for Updates' on demand without restarting Rancher Desktop?

No, there is not. I think Rancher Desktop will check once an hour on its own.
